src/argilla_server/schemas/v1/records.py

```diff
@@ -200,6 +200,21 @@ class RecordsUpdate(BaseModel):
     items: List[RecordUpdateWithId] = Field(..., min_items=RECORDS_UPDATE_MIN_ITEMS, max_items=RECORDS_UPDATE_MAX_ITEMS)
 
 
+class RecordUpsert(BaseModel):
+    fields: Optional[Dict[str, str]]
+    metadata: Optional[Dict[str, Any]]
+    external_id: Optional[str]
+    id: Optional[UUID]
```

Comment:
A small style improvement. Only moving the `id` to the top as with the rest of schemas:

```suggestion
class RecordUpsert(BaseModel):
    id: Optional[UUID]
    fields: Optional[Dict[str, str]]
    metadata: Optional[Dict[str, Any]]
    external_id: Optional[str]
```